src/main/java/it/pagopa/pn/delivery/svc/search/NotificationSearchMultiPage.java

```diff
@@ -110,15 +114,22 @@ public ResultPaginationDto<NotificationSearchRow, PnLastEvaluatedKey> searchNoti
      * @param requiredSize dimensione totale richiesta
      * @return risultati ricerca
      */
-    private ResultPaginationDto<NotificationSearchRow, PnLastEvaluatedKey> prepareGlobalResult(List<NotificationMetadataEntity> cumulativeQueryResult, int requiredSize){
+    private ResultPaginationDto<NotificationSearchRow, PnLastEvaluatedKey> prepareGlobalResult(List<NotificationMetadataEntity> cumulativeQueryResult, int requiredSize) {
 
         ResultPaginationDto<NotificationSearchRow, PnLastEvaluatedKey> globalResult = new ResultPaginationDto<>();
         globalResult.setNextPagesKey(new ArrayList<>());
 
         // tronco i risultati alla dimensione della pagina
         globalResult.setResultsPage(cumulativeQueryResult.stream()
                 .limit(inputSearchNotificationDto.getSize())
-                .map(entityToDto::entity2Dto)
+                .map(notificationMetadata ->{
+                    try {
+                        return entityToDto.entity2Dto(notificationMetadata);
+                    } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)